graph/superstep.go

```diff
@@ -0,0 +1,38 @@
+package graph
+
+import (
+	"sync"
+)
+
+type SuperStepWaiter struct {
```

Comment:
There is some argument to be made to use channels so that the path to the distributed version (i.e., with message passing) is clearer. It's okay for now to not do that though.

What I suggest instead is adding a comment in this file somewhere that basically says (1) what the process /steps here are, and (2) why this would still be possible to implement in a distributed environment. :)

Reply:
Thanks! Using channels is actually simpler :) I have updated the code.